drivers/video/video_esp32_dvp.c

```diff
@@ -378,10 +397,24 @@ static void video_esp32_cam_ctrl_init(const struct device *dev)
 	cam_ll_enable_invert_hsync(data->hal.hw, cfg->invert_hsync);
 }
 
+#ifdef CONFIG_VIDEO_ESP32_VSYNC_INTERRUPT
+static int video_esp32_enable_interrupt(const struct device *dev)
+{
+	const struct video_esp32_config *cfg = dev->config;
+	struct video_esp32_data *data = dev->data;
+
+	cam_ll_clear_interrupt_status(data->hal.hw, VIDEO_ESP32_VSYNC_MASK);
+	cam_ll_enable_interrupt(data->hal.hw, VIDEO_ESP32_VSYNC_MASK, true);
+	return esp_intr_alloc(cfg->irq_source, cfg->irq_priority,
```

Comment:
Why using the esp_intr_alloc instead of IRQ_CONNECT or IRQ_CONNECT_DYNAMIC?

Reply:
This is a good question!
I manly used it because all other espressif drivers seem to use it. 
Also I think that `esp_intr_alloc` performs some extra logic and then also calls `irq_connect_dynamic`. So It's kind of a wrapper arround `irq_connect_dynamic`?

Reply:
This is a good question!
I manly used it because all other espressif drivers seem to use it.
Also I think that `esp_intr_alloc` performs some extra logic and then also calls `irq_connect_dynamic`. So It's kind of a wrapper around `irq_connect_dynamic`?

Reply:
NVM you are right, the usage of the esp_intr alloc allows to pass extra interesting flasg and it calls the irq_connect_dynamic under its hoods.